core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/DashboardResource.scala

```diff
@@ -0,0 +1,415 @@
+        }
+
+        if (words.length == 1) {
+          // Use "*" to enable sub-string search.
+          workflowMatchQuery = workflowMatchQuery.and(
+            getSearchQuery(true, "workflow"),
+            key
+          )
+          projectMatchQuery = projectMatchQuery.and(
+            getSearchQuery(true, "project"),
+            key
+          )
+          fileMatchQuery = fileMatchQuery.and(
+            getSearchQuery(true, "file"),
+            key
+          )
+        } else {
+          // When the search query contains multiple words, sub-string search is not supported by MySQL.
+          workflowMatchQuery = workflowMatchQuery.and(
+            getSearchQuery(false, "workflow"),
+            key
+          )
+          projectMatchQuery = projectMatchQuery.and(
+            getSearchQuery(false, "project"),
+            key
+          )
+          fileMatchQuery = fileMatchQuery.and(
+            getSearchQuery(false, "file"),
+            key
+          )
+        }
+      }
+    }
+
+    // combine all filters with AND
+    var workflowOptionalFilters: Condition = noCondition()
+    workflowOptionalFilters = workflowOptionalFilters
+      // Apply creation_time date filter
+      .and(getDateFilter("creation", creationStartDate, creationEndDate, "workflow"))
+      // Apply lastModified_time date filter
+      .and(getDateFilter("modification", modifiedStartDate, modifiedEndDate, "workflow"))
+      // Apply workflowID filter
+      .and(getWorkflowIdFilter(workflowIDs))
+      // Apply owner filter
+      .and(getOwnerFilter(owners))
+      // Apply operators filter
+      .and(getOperatorsFilter(operators))
+      // Apply projectId filter
+      .and(getProjectFilter(projectIds, "workflow"))
+
+    var projectOptionalFilters: Condition = noCondition()
+    projectOptionalFilters = projectOptionalFilters
+      .and(getDateFilter("creation", creationStartDate, creationEndDate, "project"))
+      .and(getProjectFilter(projectIds, "project"))
+      // apply owner filter
+      .and(getOwnerFilter(owners))
+      .and(
+        // these filters are not available in project. If any of them exists, the query should return 0 project
+        if (
+          modifiedStartDate.nonEmpty || modifiedEndDate.nonEmpty || workflowIDs.nonEmpty || operators.nonEmpty
+        ) falseCondition()
+        else noCondition()
+      )
+
+    var fileOptionalFilters: Condition = noCondition()
+    fileOptionalFilters = fileOptionalFilters
+      .and(getDateFilter("creation", creationStartDate, creationEndDate, "file"))
+      .and(getOwnerFilter(owners))
+      .and(
+        // these filters are not available in file. If any of them exists, the query should return 0 file
+        if (
+          modifiedStartDate.nonEmpty || modifiedEndDate.nonEmpty || workflowIDs.nonEmpty || operators.nonEmpty || projectIds.nonEmpty
+        ) falseCondition()
+        else noCondition()
+      )
+
+    /**
+      *  Common attributes: 4 columns
```

Comment:
Give a brief introduction to this common schema. Also, leave some instructions for what to do after modifying the schema.

Reply:
Comments added. Please review. Thank you.